source/common/http/path_utility.cc

```diff
@@ -52,41 +52,25 @@ bool PathUtil::canonicalPath(HeaderEntry& path_header) {
   return true;
 }
 
-/* static */
 void PathUtil::mergeSlashes(HeaderEntry& path_header) {
   const auto original_path = path_header.value().getStringView();
-  if (original_path.empty()) {
-    return;
-  }
-
-  bool has_adjacent_slashes = false;
-  for (size_t i = 1; i < original_path.size(); ++i) {
-    if (original_path[i] == '/' && original_path[i - 1] == '/') {
-      has_adjacent_slashes = true;
-      break;
-    }
-    // Only operate on path component in URL.
-    if (original_path[i] == '?') {
-      break;
-    }
-  }
-  if (!has_adjacent_slashes) {
+  // Only operate on path component in URL.
+  const size_t query_start = original_path.find('?');
+  const auto path = original_path.substr(0, query_start);
```

Comment:
use explicit types in this context as type is not obvious from context unless you know string_view well.

https://google.github.io/styleguide/cppguide.html#auto